### ISSUE TITLE
Binaries are removed by El Captain upgrade

### ISSUE THREAD

**Issue by orome** [haskell/haskell-platform#221]
Following the installation of El Captain, the GHC and Cabal binary [appear to be gone](http://stackoverflow.com/q/32874888/656912). The platform needs to install them somewhere else.

**Comment by gbaz** [comment]
see also https://github.com/haskell/haskell-platform/issues/205

**Comment by erantapaa** [comment]
Question: What happens to /usr/share? Is it also locked down?

**Comment by orome** [comment]
Yes. Everything but `/usr/local` is "restricted".

**Comment by gbaz** [comment]
@orome would you be willing to test the candidate new installer as per the information on ticket #205 ?

Thanks!

**Comment by orome** [comment]
@erantapaa - I switched to Homebrew.

**Comment by gbaz** [comment]
Sure, we just need to find people on El Capitan to test this so we can release it safely to the world. If you don't have the time, no problem :-)

**Comment by gbaz** [comment]
For the record, if you had an existing platform install, and the symlinks were deleted by the El Capitan upgrade, running this script should fix it: https://github.com/mzero/haskell-platform/blob/osx-elcap-fix/hptool/os-extras/osx/bin/activate-hs

**Comment by christinem** [comment]
@gbaz I have uninstalled and re-installed Haskell using the new candidate fixed installer, and the symlinks are now properly in local/usr/bin - however, I still get the `cabal: /usr/bin/ar: permission denied` and `cabal: /usr/bin/ghc-pkg: does not exist`. I don't understand why this problem still exists after I have the new candidate installed and the symlinks should be correct.

**Comment by gbaz** [comment]
@christinem its hard to diagnose remotely here, but it sounds like there must be an old cabal somewhere in your path? perhaps in ~/.cabal/bin ?

**Comment by christinem** [comment]
@gbaz thanks for trying to assist :) interestingly, I have no bin folder in ~/.cabal. But the cabal command does work in my command line.

**Comment by gbaz** [comment]
A `whereis cabal` should reveal where the binary is coming from.

**Comment by christinem** [comment]
@gbaz Surprisingly, nothing comes up when I run that command. I deleted my earlier comment cause I thought I was confused, but yes, I don't have a bin folder in ~/.cabal, and `whereis cabal` outputs nothing.

**Comment by christinem** [comment]
The cabal binary is located in ~/Library/Haskell/bin on my machine. For extra measure, I removed ~/.cabal/bin from my path, but the `cabal: /usr/bin/ghc-pkg: does not exist` still appears. Should it be using cabal from a different location?

**Comment by gbaz** [comment]
it should be using the one from /usr/local/bin which should symlink to something inside your /Library/Frameworks/... iirc.

**Comment by christinem** [comment]
@gbaz I found the symlink, but when I try to run cabal from there, my command line completely hangs. Only the one in ~/Library/Haskell/bin works. Suffice to say I guess my testing shows that the candidate installer did not work for me.

**Comment by gbaz** [comment]
Ok, thanks for the feedback! Appreciated.

**Comment by christinem** [comment]
No problem, thanks for the help though! I'll keep tracking the issues here to see if a fix comes up.

**Comment by mgadda** [comment]
@gbaz The candidate installer was a success for me. 

I did however, have to remove `~/.cabal` and `~/Library/Haskell` entirely as well run `sudo /Library/Haskell/bin/uninstall-hs all --remove` to ensure my system was totally Haskell free. Without which, I also experienced issues with some particular version of cabal hanging as mentioned by @christinem. For posterity, my `PATH` looks like:
```
/Users/mgadda/Library/Haskell/bin 
/usr/local/bin
/usr/local/sbin
/usr/local/bin 
/usr/bin 
/bin 
/usr/sbin 
/sbin
```
